TLM/TLM/State/Options.cs

```diff
@@ -43,7 +43,7 @@ public enum PersistTo {
         /// </remarks>
         public static bool Available = false;
 
-        public static bool instantEffects = true;
+        public static bool instantEffects;
```

Comment:
I wonder why this exists.
Effects should always be instant where applicable. And non-instant should prompt a game restart or a traffic reset.

Reply:
It's also very confusingly named "Apply AI changes right away" - what does that even mean to an end-user?

It controlls this in `OptionsManager` (I have no idea what "MayPublishSegmentChanges" is supposed to mean; not investigated call sites yet):

```csharp
        public bool MayPublishSegmentChanges() {
            return Options.instantEffects && TMPELifecycle.InGameOrEditor() &&
                !TMPELifecycle.Instance.Deserializing;
        }
```

The option seems entirely pointless to me, and that `MayPublishSegmentChanges` method should be in `TMPELifecycle`.

Unfortunately it's somehow found its way in to the API. Not sure how many mods use it, but we should certainly be getting rid of it from `OptionsManager` if it's not going to break too much.

Reply:
Mark it as obsolete maybe. For future removal, or not recommended to use.

Reply:
Done:

- Removed `Options.instantEffects` as it makes no sense whatsoever
- `OptionsManager.Instance.MayPublishSegmentChanges()` marked as obsolete and redirected to new `TMPELifecycle.Instance.MayPublishSegmentChanges()` implementation